### PR TITLE
libinput-gestures: 2.76 -> 2.77

### PR DESCRIPTION
**Pull request by guttermonk**
libinput-gestures: 2.76 -> 2.77

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by momeemt** [comment]
Result of `nixpkgs-review pr 346289` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>libinput-gestures</li>
  </ul>
</details>

**Review by momeemt — CHANGES_REQUESTED** [review]
Could you change the commit message and the pr title to "libinput-gestures: 2.76 -> 2.77"? Besides, lgtm.

ref: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#:~:text=Commit%20conventions,firefox/55.0/releasenotes/

**Comment by guttermonk** [comment]
> Could you change the commit message and the pr title to "libinput-gestures: 2.76 -> 2.77"? Besides, lgtm.
> 
> ref: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#:~:text=Commit%20conventions,firefox/55.0/releasenotes/

Updated.

**Comment by guttermonk** [comment]
Possible to add "backport release-24.05" label?

**Comment by JohnRTitor** [comment]
@ofborg build libinput-gestures libinput-gestures.passthru.tests

**Review by JohnRTitor — COMMENTED** [review]
Commit message should be just

`libinput-gestures: 2.76 -> 2.77`

Remove the `update default.nix` line and force push

File: pkgs/tools/inputmethods/libinput-gestures/default.nix [hunk 1]
```diff
@@ -5,13 +5,13 @@
...
   pname = "libinput-gestures";
-  version = "2.76";
+  version = "2.77";
 
   src = fetchFromGitHub {
     owner = "bulletmark";
     repo = "libinput-gestures";
     rev = version;
-    sha256 = "sha256-Tb/gQ/2Ul4JzEiLEUPJBj9T6ZAqzMSPdgiofdnDj73Q=";
+    sha256 = "sha256-eMXNlSgQSuN+/5SXJQjsylC1ygHS87sIEmnVGFk3pzA=";
```
Comment (JohnRTitor) [inline pkgs/tools/inputmethods/libinput-gestures/default.nix:14]:
```suggestion
    hash = "sha256-eMXNlSgQSuN+/5SXJQjsylC1ygHS87sIEmnVGFk3pzA=";
```

**Comment by guttermonk** [comment]
> Commit message should be just
> 
> `libinput-gestures: 2.76 -> 2.77`
> 
> Remove the `update default.nix` line and force push

@JohnRTitor, I think I've correctly made the change you've requested. This was my first time doing a git amend so please let me know if I missed anything.

**Comment by JohnRTitor** [comment]
That unfortunately wasn't it. 

You should run `git rebase -i HEAD~2` locally, this will prompt you to make changes to 2 recent commits.
In the edit window, change the last commit's pick to squash and save. This means that your last commit will be merged to its parent commit.

This will open another edit window prompting you to edit the final commit message which you can change as you like. As I asked, keep `libinput-gestures: 2.76 -> 2.77` as the commit name and remove the rest.

This above process recreates the commits and old commits are not available at all (changes are still present don't worry, if done correctly). And so the new commits need to be force pushed. 

```
git force --push-with-lease
```

**Comment by guttermonk** [comment]
@JohnRTitor I went through the two edits but ran into some trouble on the last step:

```
 …/nixpkgs HEAD ❯ git force --push-with-lease
git: 'force' is not a git command. See 'git --help'.

 …/nixpkgs HEAD ❯ git push --force-with-lease
fatal: You are not currently on a branch.
To push the history leading to the current (detached HEAD)
state now, use

    git push origin HEAD:<name-of-remote-branch>
```
Not sure how to proceed.

**Comment by JohnRTitor** [comment]
My mistake, it's

`git push --force-with-lease`. Before that you must do `git rebase --continue`

**Comment by guttermonk** [comment]
I tried the following, but still no luck:
```
 …/nixpkgs HEAD ❯ git rebase --continue
fatal: no rebase in progress

 …/nixpkgs HEAD ❯ git rebase -i HEAD~2
Successfully rebased and updated detached HEAD.

 …/nixpkgs HEAD ❯ git rebase --continue
fatal: no rebase in progress

 …/nixpkgs HEAD ❯ git push --force-with-lease
fatal: You are not currently on a branch.
To push the history leading to the current (detached HEAD)
state now, use

    git push origin HEAD:<name-of-remote-branch>
```
Thanks again for helping me work through this.

**Comment by guttermonk** [comment]
Trying to make this change is seeming increasingly more difficult. Is having the commit message say `libinput-gestures: 2.76 -> 2.77` really required?

If so, would it be easier to make a new pull request?

**Comment by JohnRTitor** [comment]
Letme just make the change on my end

**Comment by JohnRTitor** [comment]
> Is having the commit message say `libinput-gestures: 2.76 -> 2.77` really required?
Yes it's in our CONTRIBUTING.md.

**Comment by guttermonk** [comment]
Thank you for the help. Is there any way to backport to 24.05 like I asked [above](https://github.com/NixOS/nixpkgs/pull/346289#issuecomment-2394853429)?

**Comment by JohnRTitor** [comment]
@ofborg eval

**Comment by JohnRTitor** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 346289`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>libinput-gestures</li>
  </ul>
</details>

**Comment by JohnRTitor** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 346289`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>libinput-gestures</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>libinput-gestures</li>
  </ul>
</details>

**Comment by NovaViper** [comment]
Hey I hate to bring this up it looks like the URL is dead now, updated my flake to 6a1c602a1637c3 on nixos-unstable-small and I get this error
```
trying https://github.com/bulletmark/libinput-gestures/archive/ref/tags/2.77.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
^M  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0^M  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0^M  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
^M  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (22) The requested URL returned error: 404
error: cannot download source from any mirror
```

**Comment by NovaViper** [comment]
And going to the link directly also results in a 404 screen on my end aswell

**Pull request merged**